dpnp/random/dpnp_random_state.py

```diff
@@ -125,12 +125,15 @@ def normal(self, loc=0.0, scale=1.0, size=None, dtype=None, usm_type="device"):
             else:
                 min_double = numpy.finfo('double').min
                 max_double = numpy.finfo('double').max
-                if (loc >= max_double or loc <= min_double) and dpnp.isfinite(loc):
+
+                # TODO: switch to dpnp.isfinite() and dpnp.signbit() once functions are available,
+                # but for now use direct numpy calls without call_origin() wrapper, since data is a scalar
+                if (loc >= max_double or loc <= min_double) and numpy.isfinite(loc):
                     raise OverflowError(f"Range of loc={loc} exceeds valid bounds")
 
-                if (scale >= max_double) and dpnp.isfinite(scale):
+                if (scale >= max_double) and numpy.isfinite(scale):
```

Comment:
Perhaps factor out uses of `numpy.isfinite` into your own routine which can be modified as needed. 

If `scale` always remains a Python or NumPy scalar, than not compute-follows-data violation occurs, but if `scale` can be an array, then your modular routine can be modified to apply appropriate validation function.

Reply:
Added internal functions _is_finite_scalar() and _is_finite_scalar() to dpnp_random_state.py with TODO comment inside to replace with dpnp functionality once availbale.